src/intrprtr.c

```diff
@@ -4682,7 +4682,22 @@ void            IntrInfoBegin( void )
 
 }
 
-Obj             InfoDecision;
+static Obj InfoDecision;
+static Obj InfoDecisionFast;
+static Obj IsInfoClassListRep;
+
+Obj InfoCheckLevel(Obj selectors, Obj level)
+{
+    if (IS_POS_INTOBJ(level) &&
+        CALL_1ARGS(IsInfoClassListRep, selectors) == True) {
+        Obj index = ELM_PLIST(selectors, 1);
+        return CALL_3ARGS(InfoDecisionFast, index, selectors, level);
```

Comment:
The main reason is in HPC-GAP I have to take a lock, and I wasn't sure how to do that.

Once GAPDoc is moved to use the new methods in this PR, I will re-write the internal data structures to be (a) more efficient, and (b) easier to lock. Then the code will be entirely in C.